cai_causal_graph/identify_utils.py

```diff
@@ -387,3 +387,45 @@ def identify_markov_boundary(graph: Union[CausalGraph, Skeleton], node: NodeLike
         mb = graph.get_neighbors(node_id)
 
     return mb
+
+
+def identify_colliders(graph: CausalGraph, unshielded: bool = False) -> List[str]:
+    """
+    Identify all the collider nodes in the provided `graph`.
+
+    :param graph: The graph given by a `cai_causal_graph.causal_graph.CausalGraph` instance.
+    :param unshielded: If `True`, only unshielded colliders are returned. If `False`, all colliders are returned.
```

Comment:
would be nice to also include a brief definition about what "unshielded" means. #nitpick